src/CoreBundle/Bridge/Chameleon/Module/Breadcrumb/BreadcrumbBackendModule.php

```diff
@@ -0,0 +1,323 @@
+<?php
+
+/*
+ * This file is part of the Chameleon System (https://www.chameleonsystem.com).
+ *
+ * (c) ESONO AG (https://www.esono.de)
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ChameleonSystem\CoreBundle\Bridge\Chameleon\Module\Breadcrumb;
+
+use ChameleonSystem\CoreBundle\DataAccess\MenuItemDataAccessInterface;
+use ChameleonSystem\CoreBundle\DataModel\BackendBreadcrumbItem;
+use ChameleonSystem\CoreBundle\DataModel\MenuCategoryAndItem;
+use ChameleonSystem\CoreBundle\Service\LanguageServiceInterface;
+use ChameleonSystem\CoreBundle\Util\UrlUtil;
+use IMapperCacheTriggerRestricted;
+use IMapperVisitorRestricted;
+use Symfony\Component\Debug\Exception\UndefinedMethodException;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Translation\TranslatorInterface;
+
+class BreadcrumbBackendModule extends \MTPkgViewRendererAbstractModuleMapper
+{
+    /**
+     * @var MenuItemDataAccessInterface
+     */
+    private $menuItemDataAccess;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var UrlUtil
+     */
+    private $urlUtil;
+
+    /**
+     * @var LanguageServiceInterface
+     */
+    private $languageService;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    public function __construct(
+        MenuItemDataAccessInterface $menuItemDataAccess,
+        RequestStack $requestStack,
+        UrlUtil $urlUtil,
+        LanguageServiceInterface $languageService,
+        TranslatorInterface $translator
+    ) {
+        parent::__construct();
+
+        $this->menuItemDataAccess = $menuItemDataAccess;
+        $this->requestStack = $requestStack;
+        $this->urlUtil = $urlUtil;
+        $this->languageService = $languageService;
+        $this->translator = $translator;
+    }
+
+    // NOTE caching is difficult here because cache delete triggers would need to include virtually every table
+    //   as any entry's name might end up in the breadcrumb.
```

Comment:
the comment should be added to the allowCache method

Reply:
Added/moved